### PR TITLE
Alert queue optimizations - v3

### PR DESCRIPTION
**Pull request by jufajardini**
Previous PR: https://github.com/OISF/suricata/pull/7290

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/4943

Describe changes:
- rebased last PR, to have the CI checks run

suricata-verify-pr: 808
https://github.com/OISF/suricata-verify/pull/808

### CODE REVIEW

**Comment by victorjulien** [comment]
QA on my end looks good.

**Comment by suricata-qa** [comment]
Note: please make sure all tickets in the commits are mentioned in the PR:
- https://redmine.openinfosecfoundation.org/issues/5179
- https://redmine.openinfosecfoundation.org/issues/4207

Please update PR body and remove the 'needs ticket' label when done.

File: src/detect.c [hunk 1]
```diff
@@ -1019,6 +1103,21 @@ DetectRunTxSortHelper(const void *a, const void *b)
         return s0->id > s1->id ? 1 : -1;
 }
 
+/** \internal
+ * \brief sort helper for sorting alerts by priority
+ *
+ * The id field is set from Signature::num, so we have higher priority (lower ids) first
```
[the remark this answers is not in the document]
Comment (jufajardini) [reply src/detect.c:1109]: This actually needed some rewording, thanks for making me re-read it :P

**Comment by jufajardini** [comment]
Followed by https://github.com/OISF/suricata/pull/7314

**Pull request closed without merging**